rfc/rfc006-distributed-registry.md

```diff
@@ -322,7 +327,66 @@ The care organization refers to it:
 }
 ```
 
-### 4.1. Contact information
+### 4.1 Service references
+
+Any `serviceEndpoint` with a value that starts with `did` is a reference to another service.
+As specified earlier, all references use the `type` query parameter.
+The origin of a reference is from the `serviceEndpoint` context, while the target is a complete `service` object.
+When a reference targets a `service`, it actually targets the `serviceEndpoint` of the service.
+
+A client doesn't care about the references. When a client queries for particular services, the references need to be resolved.
+Resolving a reference might lead to resolving more references and may lead to an infinite loop.
+Resolving a reference MUST NOT go deeper than 5 levels. Given 5 services, A through E, A references B, B references C, etc. Given the maximum depth of 5, E MUST contain an absolute endpoint URI.
+
+Another limitation is that a reference that originates from a compound service MUST NOT resolve to another compound service. When resolving that would lead to a map of URIs nested within a map of URIs.
```

Comment:
Drop "When"

If you specify the effect, ("resolving that would lead to a map..." etc), specify the problem as well